test/query_component_test.dart

```diff
@@ -470,61 +500,55 @@ void main() {
-                ioId1,
-                startDate,
-                any))
-            .thenAnswer(
-                (_) => Future.value(reportingCreateQueryApiJsonResponse));
-        when(mockQueryService.execReportingGetQuery(queryId))
-            .thenAnswer((_) => Future.value(reportingGetQueryApiJsonResponse));
-        when(mockQueryService.execReportingDownload(downloadLink))
-            .thenAnswer((_) => reportCompleter.future);
+        final answers = [
+          Future.value(singleIOJsonString),
+          Future.value(reportingCreateQueryApiJsonResponse),
+          Future.value(reportingGetQueryApiJsonResponse),
+          reportCompleter.future,
+        ];
+
+        var callCount = 0;
+        when(mockGoogleApiDart.request(any))
+            .thenAnswer((_) => answers[callCount++]);
 
         expectedInput = [
           generateInsertionOrder(
               advertiserId, mediaPlanId1, ioId1, startDate, endDate, spending1)
         ];
       });
 
-      tearDown(() => clearInteractions(mockQueryService));
+      tearDown(() => clearInteractions(mockGoogleApiDart));
 
       test('no advertiser id and no io id disables the populate button',
           () async {
         await queryComponentPO.clickPopulate();
-        verifyNever(mockQueryService.execDV3Query(any, any, any, any, any));
+        verifyNever(mockGoogleApiDart.request(any));
       });
 
       test('no advertiser id disables the populate button', () async {
         await queryComponentAccordionPO.typeInsertionOrderId(ioId1);
         await queryComponentPO.clickPopulate();
-        verifyNever(mockQueryService.execDV3Query(any, any, any, any, any));
+        verifyNever(mockGoogleApiDart.request(any));
       });
 
       test('non-integer advertiser id disables the populate button', () async {
         await queryComponentAccordionPO.typeAdvertiserId('wrong-format-id');
         await queryComponentAccordionPO.typeInsertionOrderId(ioId1);
         await queryComponentPO.clickPopulate();
-        verifyNever(mockQueryService.execDV3Query(any, any, any, any, any));
+        verifyNever(mockGoogleApiDart.request(any));
       });
 
       test('no io id disables the populate button', () async {
         await queryComponentAccordionPO.typeAdvertiserId(advertiserId);
         await queryComponentPO.clickPopulate();
-        verifyNever(mockQueryService.execDV3Query(any, any, any, any, any));
+        verifyNever(mockGoogleApiDart.request(any));
       });
 
       test('non-integer io id disables the populate button', () async {
         await queryComponentAccordionPO.typeAdvertiserId(advertiserId);
         await queryComponentAccordionPO.typeInsertionOrderId('wrong-format-id');
         await queryComponentPO.clickPopulate();
-        verifyNever(mockQueryService.execDV3Query(any, any, any, any, any));
+        verifyNever(mockGoogleApiDart.request(any));
       });
 
       test('highlighting unchecked invokes populate(single-IO, false)',
```

Comment:
no need to test every combination of query type + checkbox
1 test per query type, and 1 test for each checkbox state (i.e. 2)